### PR TITLE
Fix DeprecationWarning: pkg_resources is deprecated as an API

### PR DESCRIPTION
**Pull request by valentino-sm**

### CODE REVIEW

**Comment by insolor** [comment]
@valentino-sm I've dropped Python < 3.8 already, so code can be simplified

File: pymorphy2/analyzer.py [hunk 1]
```diff
@@ -102,18 +102,17 @@ def apply_to_tags(self, word, word_lower, tags):
...
+    ep = entry_points()
+    result: set[EntryPoint] = set()
+    if not hasattr(ep, 'select'):
+        # Python < 3.10 old entry_points API
+        for group in args:
+            result.update(ep.get(group) or ())
+    else:
+        for group in args:
+            result.update(ep.select(group=group))
+    return list(result)
```
Comment (insolor) [inline pymorphy2/analyzer.py:115]: It's not necessary to convert the result into a list. The result is used as an iterable object in a for loop (in `_lang_dict_paths` function below). So, `set` is quite suitable.

**Review by insolor — CHANGES_REQUESTED** [review]

File: pymorphy2/analyzer.py [hunk 2]
```diff
@@ -102,18 +102,17 @@ def apply_to_tags(self, word, word_lower, tags):
 
 
 def _iter_entry_points(*args, **kwargs):
-    """ Like pkg_resources.iter_entry_points, but uses a WorkingSet which
```
[the remark this answers is not in the document]
Comment (valentino-sm) [reply pymorphy2/analyzer.py]: Could you tell me the correct docstring, that should be good.

Thank you for your support and maintain the project❤️

File: pymorphy2/analyzer.py [hunk 3]
```diff
@@ -111,9 +111,17 @@ def _iter_entry_points(*args, **kwargs):
...
-    import pkg_resources
-    ws = pkg_resources.WorkingSet()
-    return ws.iter_entry_points(*args, **kwargs)
+    from importlib.metadata import entry_points, EntryPoint
+    ep = entry_points()
+    result: set[EntryPoint] = set()
+    if not hasattr(ep, 'select'):
+        # Python < 3.10 old entry_points API
+        for group in args:
+            result.update(ep.get(group) or ())
```
Comment (insolor) [inline pymorphy2/analyzer.py:120]: `ep.get(group) or ()` can be changed to `ep.get(group, ())`

File: pymorphy2/analyzer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (insolor) [reply pymorphy2/analyzer.py]: Something like that:
```
"""
Finds entry points of installed packages, including ones installed
after the current process is started.

The main use case is to make ``!pip install pymorphy2-fork`` work
within a Jupyter or Google Colab notebook.
See https://github.com/kmike/pymorphy2/issues/131
"""
```

File: pymorphy2/analyzer.py (hunk 3 above)
Comment (valentino-sm) [reply pymorphy2/analyzer.py]: Of course you are right, this is only for strict type checking with a return value of `Entrypoints | None`

**Comment by insolor** [comment]
Thank you for the contribution 👍🏻

**Comment by BLKSerene** [comment]
@valentino-sm Hi, would you mind if I port this PR to https://github.com/no-plagiarism/pymorphy3?

**Comment by valentino-sm** [comment]
> @valentino-sm Hi, would you mind if I port this PR to https://github.com/no-plagiarism/pymorphy3?

Sure!
Thank you everyone for your work

**Pull request merged**